### PR TITLE
Perform missing null check for missing response fields.

### LINKED ISSUE #43: PlaceDetails.fromJson - check for missing response field
Defect,  google api allows you to get place details with selective field as they charge based on the information you access.  The fromJson method should do appropriate check for null and empty before assigning the value to object.

### PR DESCRIPTION
**Pull request by pratikpparikh**
This pull request is to fix #43 around places.

### CODE REVIEW

**Comment by pratikpparikh** [comment]
@lejard-h could you please review this code?

File: lib/src/places.dart [hunk 1]
```diff
@@ -389,10 +389,12 @@ class PlacesSearchResponse extends GoogleResponseList<PlacesSearchResult> {
           json['status'],
           json['error_message'],
           json['results']
-              .map((r) => PlacesSearchResult.fromJson(r))
-              .toList()
-              .cast<PlacesSearchResult>(),
-          (json['html_attributions'] as List).cast<String>(),
+              ?.map((r) => PlacesSearchResult.fromJson(r))
+              ?.toList()
+              ?.cast<PlacesSearchResult>(),
```
Comment (lejard-h) [inline lib/src/places.dart]: Could you also results to be not null ? `.cast<PlacesSearchResult>() ?? []`

File: lib/src/places.dart [hunk 2]
```diff
@@ -564,19 +568,21 @@ class PlaceDetails {
   factory PlaceDetails.fromJson(Map json) => json != null
       ? PlaceDetails(
           json['address_components']
-              .map((addr) => AddressComponent.fromJson(addr))
-              .toList()
-              .cast<AddressComponent>(),
+              ?.map((addr) => AddressComponent.fromJson(addr))
+              ?.toList()
+              ?.cast<AddressComponent>(),
```
Comment (lejard-h) [inline lib/src/places.dart]: same thing here, pass an empty list when null

File: lib/src/places.dart [hunk 3]
```diff
@@ -783,9 +798,9 @@ class PlacesAutocompleteResponse extends GoogleResponseStatus {
           json['status'],
           json['error_message'],
           json['predictions']
-              .map((p) => Prediction.fromJson(p))
-              .toList()
-              .cast<Prediction>())
+              ?.map((p) => Prediction.fromJson(p))
+              ?.toList()
+              ?.cast<Prediction>())
```
Comment (lejard-h) [inline lib/src/places.dart]: same thing, empty list when null

**Pull request merged**